src/controllers/share.controller.ts

```diff
@@ -0,0 +1,98 @@
+import { Request, Response, Router, NextFunction } from 'express';
+import { v4 as uuidv4 } from 'uuid';
+import mongoose from 'mongoose';
+import { Controller } from '../interfaces';
+import { AsyncAPIDocument } from '@asyncapi/parser';
+import { validationMiddleware } from '../middlewares/validation.middleware';
+import Data from '../models/data';
+import { ProblemException } from '../exceptions/problem.exception';
+import { logger } from '../utils/logger';
+export class ShareController implements Controller {
+  public basepath = '/share';
+
+  private async initializeDatabase() {
+    await mongoose
+      .connect(process.env.MONGODB_URI || 'mongodb://localhost:27017/asyncapi')
+      .then(() => {
+        logger.info('🚀 Database connection is successful');
+      })
+      .catch((err) => {
+        logger.error(`Database connection failed${err.message}`);
+      });
+  }
+
+  private async share(req: Request, res: Response, next: NextFunction) {
+    await this.initializeDatabase();
+    const stringifiedSpec = AsyncAPIDocument.stringify(
+      req.asyncapi?.parsedDocument
+    );
+    const docId = uuidv4();
+    try {
+      const data = new Data({
+        doc: stringifiedSpec,
+        docId,
+        date: Date.now(),
+      });
+      await data.save();
+      res.status(201).json({
+        url: `https://studio.asyncapi.com/${docId}`,
+      });
```

Comment:
You should return only id, not whole url. Someone can use that id in another app.